logstash-core/lib/logstash/runner.rb

```diff
@@ -9,7 +9,10 @@
 LogStash::OSS = ENV["OSS"] == "true" || !File.exists?(LogStash::XPACK_PATH)
 
 if !LogStash::OSS
-  $LOAD_PATH << File.join(LogStash::XPACK_PATH, "lib")
+  xpack_dir = File.join(LogStash::XPACK_PATH, "lib")
+  unless $LOAD_PATH.include?(xpack_dir)
```

Comment:
We were repeatedly adding x-pack here in the tests because the x-pack path was already loaded, made the path adjustment conditional to be a little more robust here between test and production code